internal/abi/fastly/hostcalls_guest.go

```diff
@@ -2048,3 +2048,119 @@ func GeoLookup(ip net.IP) ([]byte, error) {
 
 	return buf.AsBytes(), nil
 }
+
+// witx:
+//
+//   (module $fastly_object_store
+//	   (@interface func (export "open")
+//	     (param $name string)
+//	     (result $err (expected $object_store_handle (error $fastly_status)))
+//	  )
+
+//go:wasm-module fastly_object_store
+//export open
+//go:noescape
+func fastlyObjectStoreOpen(
+	name prim.Wstring,
+	h *objectStoreHandle,
+) FastlyStatus
+
+// ObjectStore represents a Fastly object store, a collection of key/value pairs.
+// For convenience, keys and values are both modelled as Go strings.
+type ObjectStore struct {
+	h objectStoreHandle
+}
+
+// ObjectStoreOpen returns a reference to the named object store, if it exists.
+func OpenObjectStore(name string) (*ObjectStore, error) {
+	var o ObjectStore
+
+	if err := fastlyObjectStoreOpen(
+		prim.NewReadBufferFromString(name).Wstring(),
+		&o.h,
+	).toError(); err != nil {
+		return nil, err
+	}
+
+	return &o, nil
+}
+
+// witx:
+//
+//   (@interface func (export "lookup")
+//	   (param $store $object_store_handle)
+//	   (param $key string)
+//	   (param $body_handle_out (@witx pointer $body_handle))
+//	   (result $err (expected (error $fastly_status)))
+//	)
+
+//go:wasm-module fastly_object_store
+//export lookup
+//go:noescape
+func fastlyObjectStoreLookup(
+	h objectStoreHandle,
+	key prim.Wstring,
+	b *bodyHandle,
+) FastlyStatus
+
+// Lookup returns the value for key, if it exists.
+func (o *ObjectStore) Lookup(key string) (string, error) {
+
+	body := HTTPBody{}
+
+	if err := fastlyObjectStoreLookup(
+		o.h,
+		prim.NewReadBufferFromString(key).Wstring(),
+		&body.h,
+	).toError(); err != nil {
+		return "", err
+	}
+
+	b, err := io.ReadAll(&body)
+	if err != nil {
+		return "", err
+	}
+
+	return string(b), nil
```

Comment:
@dgryski Am I correct that this is reading the Object Store value into memory and then parsing it into a string? If that’s the case then I think it won’t be possible to lookup large objects as they could exceed the allowed max memory on C@E. If possible, it'd be nice to have an option which returns a stream or iterator

Reply:
Hmm.. good point.  I was going for a simple API, but I'll switch this out a returning an `io.Reader` instead to allow streaming.

Reply:
The same thing applies to `Insert`.

Reply:
@joeshaw Yeah, I was doing to do both sides.